pkg/collector/corechecks/cluster/orchestrator/transformers/k8s/pod_test.go

```diff
@@ -1225,3 +1225,75 @@ func TestConvertNodeSelectorRequirements(t *testing.T) {
 		})
 	}
 }
+
+func TestExtractPodResourceRequirementsSidecar(t *testing.T) {
+	restartAlways := v1.ContainerRestartPolicy("Always")
+	tests := map[string]struct {
+		input    []v1.Container
+		expected []*model.ResourceRequirements
+	}{
+		"sidecar pod": {
```

Comment:
could be interesting to add
- a test here for a sidecar pod without resource requirements
- test in TestExtractPod of a full pod with a native sidecar container